### ISSUE TITLE
Making custom data available in Results page

### ISSUE THREAD

**Issue by jjgao** [cBioPortal/cbioportal#7619]
One common use case of OncoPrinter is adding additional tracks of clinical data to existing samples in the portal. It would reduce ease the task if we can support custom data in OncoPrint directly.

Custom data can be added to the study view page through "Charts => Custom Data." It would be useful to make the custom data available in in OncoPrint.

The custom data can also be useful in Plots tab. 

We also have clinical data in Comparison tab but since the calculation is on backend so it will not be possible to add the custom data there.

- [x] adding categorical data to oncoprint
- [ ] adding categorical data to Plots tab
- [ ] adding numerical data to oncoprint
- [ ] adding numerical data to Plots

@cBioPortal/product

**Comment by adamabeshouse** [comment]
@jjgao interesting idea. we can make it another tab in the add clinical tracks dropdown. why did you tag session-service, we're going to save the data to the session?

**Comment by jjgao** [comment]
I think the custom data is saved in session service. @kalletlak please correct me if I am wrong.

**Comment by adamabeshouse** [comment]
some questions:
(1) this is via copy+paste, or uploading files?
(2) whats the format?
(3) is there a way to share custom data with other users?

@jjgao @kalletlak

**Comment by jjgao** [comment]
@adamabeshouse the data should be added from study view through custom data. Sharing is a separate issue.

I've created an epic for the related issues: https://github.com/cBioPortal/cbioportal/issues/7728

**Comment by jjgao** [comment]
@adamabeshouse what's left for this?

**Comment by adamabeshouse** [comment]
I merged [the PR for categorical custom data](https://github.com/cBioPortal/cbioportal-frontend/pull/3326) but I think we were still looking to do this with numerical data, waiting on some work from @kalletlak if I remember correctly though I honestly lost track of this

**Comment by jjgao** [comment]
Thanks, @adamabeshouse . I added a checklist above.
